src/deadline_worker_agent/sessions/actions/scripts/attachment_download.py

```diff
@@ -2,49 +2,162 @@
 
 #! /usr/bin/env python3
 import argparse
+import json
 import time
+import os
 import boto3
+from typing import Dict, List
 
-from deadline.job_attachments import api
+from deadline.job_attachments.download import download_files_from_manifests
+from deadline.job_attachments.asset_manifests.decode import decode_manifest
+from deadline.job_attachments.asset_manifests import BaseAssetManifest
+from deadline.job_attachments.models import JobAttachmentS3Settings
+from deadline.job_attachments.progress_tracker import DownloadSummaryStatistics
+from deadline_worker_agent.sessions.attachment_models import WorkerManifestProperties
+from deadline_worker_agent.aws.deadline import (
+    record_sync_inputs_fail_telemetry_event,
+    record_sync_inputs_telemetry_event,
+)
 
-"""
-A small script to download job output using attachment download.
-This is available in deadline-cloud as python API and AWS Deadline Cloud CLI.
 
-Example usage:
+def load_worker_manifest_properties(worker_properties_file: str) -> List[WorkerManifestProperties]:
+    """
+    Load and parse worker manifest properties from a JSON file.
 
-python attachment_download.py \
-    -pm /sessions/session-f63c206fb5f04c04aa17821001aa3847fajfm5x4/path_mapping.json \
-    -s3 s3://test-job-attachment/DeadlineCloud \
-    -m /sessions/session-f63c206fb5f04c04aa17821001aa3847fajfm5x4/manifests/0bb7eb91fdf8780c4a7e6174de6dfc5e_manifest \
-    -m /sessions/session-f63c206fb5f04c04aa17821001aa3847fajfm5x4/manifests/0bb7eb91fdf8780c4a7e6174de6dfc5e_manifest
-"""
+    Args:
+        worker_properties_file: Path to the worker properties JSON file
 
+    Returns:
+        List of WorkerManifestProperties objects
 
-def download(s3_root_uri: str, path_mapping_rules: str, manifests: list[str]) -> None:
-    api.attachment_download(
-        manifests=manifests,
-        s3_root_uri=s3_root_uri,
-        boto3_session=boto3.session.Session(),
-        path_mapping_rules=path_mapping_rules,
-    )
+    Raises:
+        FileNotFoundError: If the worker properties file doesn't exist
+        json.JSONDecodeError: If the file contains invalid JSON
+        ValueError: If the JSON structure is invalid
+    """
+    worker_manifest_properties = []
 
+    with open(worker_properties_file, "r") as f:
+        worker_manifest_properties_data = json.load(f)
 
-if __name__ == "__main__":
+    for item in worker_manifest_properties_data or []:
+        worker_prop = WorkerManifestProperties.from_dict(item)
+        worker_manifest_properties.append(worker_prop)
+
+    return worker_manifest_properties
```

Comment:
nit: We can simplify this after ensuring `worker_manifest_properties_data` is a list of dicts

```suggestion
    return [WorkerManifestProperties.from_dict(item) for item worker_manifest_properties_data]
```